web/src/features/charts/graphUtils.test.ts

```diff
@@ -84,3 +90,51 @@ describe('getElectricityProductionValue', () => {
     expect(actual).toEqual(null);
   });
 });
+
+describe('getTotalElectricity', () => {
+  const zoneData = {
+    totalCo2Production: 100,
+    totalCo2Discharge: 50,
+    totalProduction: 200,
+    totalDischarge: 100,
+    totalImport: 50,
+    totalCo2Import: 25,
+  } as ZoneDetail;
+
+  it('handles emissions for consumption', () => {
+    const actual = getTotalElectricity(zoneData, true, Mode.CONSUMPTION);
+    expect(actual).toEqual(175);
+  });
+
+  it('handles power for consumption', () => {
+    const actual = getTotalElectricity(zoneData, false, Mode.CONSUMPTION);
+    expect(actual).toEqual(350);
+  });
+
+  it('handles emissions for production', () => {
+    const actual = getTotalElectricity(zoneData, true, Mode.PRODUCTION);
+    expect(actual).toEqual(150);
+  });
+
+  it('handles power for production', () => {
+    const actual = getTotalElectricity(zoneData, false, Mode.PRODUCTION);
+    expect(actual).toEqual(300);
+  });
+
+  it('returns NaN when productionValue is null', () => {
+    const actual = getTotalElectricity(
+      { ...zoneData, totalProduction: Number.NaN },
```

Comment:
Will do a full review and testing tomorrow but until then:
Shouldn't this be Null if it tests that that it should return Nan if it's null? 😉
```suggestion
      { ...zoneData, totalProduction: Null},
```